pkg/api/unversioned/types.go

```diff
@@ -54,6 +54,38 @@ type ListMeta struct {
 	ResourceVersion string `json:"resourceVersion,omitempty"`
 }
 
+// ListOptions is the query options to a standard REST list/watch calls.
+type ListOptions struct {
+	TypeMeta `json:",inline"`
+
+
+	// FIXME: It would be super convinient to have LabelSelector and FieldSelector
+	// of types: labels.Selector and fields.Selector respectively (no additional parsing
+	// in random places of the code).
+	//
+	// However, this would require (at least);
+	// - adding support for serialization/deserialization labels/fields.Selector to/from JSON
+	//   [my feeling is that MarshalJSON and UnmarshalJSON is what needs to be implemented for them]
+	// - adding some special treatment of those fields in convert to URL params - this would be
+  //   a dedicated, non-generic code (I think).
```

Comment:
> This will not work for current implementation (i.e. it will not work for url.Value).
However, after looking into this code, I think that it should be possible to changing the code converting to/from url.Value to make it work.
@lavalamp - so you think that having those fields as "labels/fields.Selector" (which mean different that in v1 api) is ok, right?

Right. The thing that matters for compatibility is wire format, not the go types.

> I don't get it - are we still talking about ListOptions or something more generic? If the former - I can't see usecase for different list options in different apigroup. And if we need different options for different versions - we shouldn't make then unversioned, right?
So to summarize - I'm a bit lost after this comment.

In the interest of making progress, I'm going to say to put it in unversioned for now. We are establishing machinery for all option-type things, and the machinery should work no matter where the structs are located.